### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by spheromak**
merge Ryanfrantz/master  from #86  This is a merge of that work, to get CI/cycle working on a branch before we pull into master.

---
This change is [Reviewable](https://reviewable.io/reviews/go-chef/chef/96)

### CODE REVIEW

**Comment by MarkGibbons** [comment]
The Groups.List function doesn't work as is.

**Comment by spheromak** [comment]
LGTM will try to work out whats up in CI this week if i have time.

**Comment by MarkGibbons** [comment]
I fixed a bunch of small error formatting problems that made go test fail.  I changed the return from the groups List function to be a map instead of an array.  The tests for the project pass now when I run them.

File: sandbox_test.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func TestSandboxesPost(t *testing.T) {
 
 	// post the new sums/files to the sandbox
 	_, err := client.Sandboxes.Post(sums)
-	if err != nil {
+	if err != nil
```
Comment (MarkGibbons) [inline sandbox_test.go]: Typo

**Pull request merged**